examples/joins/README.md

```diff
@@ -1,193 +1,231 @@
+
+* `purchases` - Each purchase info is kept in a file named by concatenating the purchase's order number and its store ID.
+```shell
+    └── ORDERW080520_STOREID1.txt
+    └── ORDERW080521_STOREID1.txt
+    └── ORDERW078929_STOREID2.txt
+    └── ...
+```
+* `returns` - Same naming convention as purchases.
+```shell
+    └── ORDERW080528_STOREID5.txt
+    └── ODERW080520_STOREID1.txt
+    └── ...
+```
+## 3. Data preparation
+Let's first create our mock dataset and create/populate our repositories.
+This preparatory step will allow you to experiment with the inner and outer join examples in any order.
+
+***Step 1*** - Prepare your data
+The setup target of the `Makefile` in `pachyderm/examples/joins` will create 3 directories (stores, purchases, and returns) with our example data.
+In the `examples/joins` directory, run:
+```shell
+$ make setup
+```
+***Step 2*** - Create and populate Pachyderm's repositories.
+In the `examples/joins` directory, run:
+```shell
+$ make create
+```
+or run:
+```shell
+$ pachctl create repo stores
+$ pachctl create repo purchases
+$ pachctl create repo returns
+$ pachctl put file -r stores@master:/ -f stores
+$ pachctl put file -r purchases@master:/ -f purchases
+$ pachctl put file -r returns@master:/ -f returns
+```
+
+Have a quick look at your repositories: 
+```shell
+$ pachctl list file stores@master
+$ pachctl list file purchases@master
+$ pachctl list file returns@master
+```
+You should see the following files:
+- Stores:
+
+![stores_repository](./img/pachctl_list_file_stores_master.png)
+
+- Purchases:
+
+![purchases_repository](./img/pachctl_list_file_purchases_master.png)
+
+- Returns:
+
+![returns_repository](./img/pachctl_list_file_returns_master.png)
+
+## 4. Example 1 - An Inner-Join pipeline creation 
+***Goal***
+List all purchases by zip code. 
+>![pach_logo](./img/pach_logo.svg) Remember that we will only see the zip code of the stores where purchases actually happened. That is what inner joins do.
+
+1. **Pipeline input repositories**: `stores` and `purchases` - Inner join by STOREID.
+2. **Pipeline**: Executes a python code reading the `zipcode` in the matching STOREIDx.txt file and appending the matched purchase file's content to a text file named after the zip code.
+3. **Pipeline output repository**: `inner_join`- list of text files named after the stores' zip codes in which purchases have been made and containing the list of said purchases. 
+
+In the diagram below, we have mapped out the data of our example. We will expect the 2 following files in the output repository of our `inner_join` pipeline: 
+```shell
+    └── 02108.txt
+    └── 90210.txt
+```
+Each should contain 3 purchases.
+
+![inner_join](./img/inner_join.png)
+
+***Step 3*** - Now let's create your pipeline. 
+
+Because unprocessed data are awaiting in your entry repositories, the pipeline creation will automatically trigger a job.
+In the `examples/joins` directory, run:
+```shell
+$ make inner-join
```

Comment:
IMO, having a make target when its one line is a little weird. It's fine to leave in the makefile, when it's only one line, I think it's better for the tutorial to show the user the pachctl cmd (they'll learn more) rather than the make target as the primary/preferred cmd to run. The make is better when we're scripting a bunch of setup.

Reply:
You are 100% right. I restructured the makefile to make sure that anyone could run both examples in any order without running into a repo that has already been created and ended up with this one line. Makes no sense. Removing it. Thx for the feedback.